tests/test_tools_pr_comments.py

```diff
@@ -0,0 +1,86 @@
+# Tests for functions defined in 'tools/pr_comments.py' of the EESSI
+# build-and-deploy bot, see https://github.com/EESSI/eessi-bot-software-layer
+#
+# The bot helps with requests to add software installations to the
+# EESSI software layer, see https://github.com/EESSI/software-layer
+#
+# author: Thomas Roeblitz (@trz42)
+#
+# license: GPLv2
+#
+
+# Standard library imports
+
+# Third party imports (anything installed into the local Python environment)
+import pytest
+from unittest.mock import patch
+
+# Local application imports (anything from EESSI/eessi-bot-software-layer)
+from tools.pr_comments import get_comment, get_submitted_job_comment
+
+
+class MockIssueComment:
+    def __init__(self, body):
+        self.body = body
+
+
+@pytest.fixture
+def raise_exception():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.side_effect = Exception()
+        instance.get_issue_comments.return_value = ()
+        yield instance
+
+
+@pytest.fixture
+def pr_no_comments():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.return_value = ()
+        yield instance
+
+
+@pytest.fixture
+def pr_single_comment():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.return_value = (MockIssueComment("foo"),)
+        yield instance
```

Comment:
Maybe I'm missing something, but are we actually still testing `get_issue_comments` here?
Is any code from that function still being run?
We always override the return value, so we're basically only testing whether the body of `get_issue_comments` triggers an exception somehow. We're definitely not testing whether `get_issue_comments` returns what it should be returning...

It would be better to mock whatever function/method is used by `get_issue_comments` to steer it towards returning `foo`, but not fiddle with the return value of `get_issue_comments`?

Reply:
Ah, wait, we're actually testing our `get_comment` function here, which calls `get_issue_comments` (which is provided by `PyGithub`, not by us).

OK, then this does make sense, sorry :)